core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```
@@ -555,6 +559,31 @@ class HttpGoogleServicesDAO(val clientSecrets: GoogleClientSecrets,
     }
   }
 
+  override def isBillingAccountEnabled(billingAccount: RawlsBillingAccountName)(implicit
+    executionContext: ExecutionContext
+  ): Future[Boolean] =
+    getBillingAccountEnabled(billingAccount, getBillingServiceAccountCredential)
+
+  protected def getBillingAccountEnabled(billingAccount: RawlsBillingAccountName, credential: Credential)(implicit
+    executionContext: ExecutionContext
+  ): Future[Boolean] = {
+    implicit val service = GoogleInstrumentedService.Billing
+    val fetcher = getCloudBillingManager(credential).billingAccounts().get(billingAccount.value)
+    retryWithRecoverWhen500orGoogleError { () =>
+      val response = blocking {
+        executeGoogleRequest(fetcher)
+      }
+      response.getOpen.booleanValue()
+    } { case e: GoogleJsonResponseException =>
+      throw new RawlsExceptionWithErrorReport(
+        ErrorReport(
+          StatusCodes.InternalServerError,
```

Could this try to propagate the status code that Google threw instead of always returning a 500? See  https://github.com/broadinstitute/rawls/blob/7180e1d969529bf77affc982f19dc0efe63175ec/core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpSamDAO.scala#L113 as an example.